app/modules/users/components/user_profile_buttons.svelte

```diff
@@ -0,0 +1,204 @@
+    app.execute('show:models:on:map', [ userModel, app.user ])
+  }
+</script>
+
+<div class="profile-buttons">
+  {#if mainUser}
+    <a
+      class="editProfile action tiny-button light-blue"
+      href="/settings/profile"
+      on:click={loadInternalLink}
+    >
+      {@html icon('pencil')}
+      {I18n('edit profile')}
+    </a>
+    <a
+      class="addItems action tiny-button light-blue"
+      href="/add/scan" title={I18n('title_add_layout')}
+      on:click={loadInternalLink}
+    >
+      {@html icon('plus')} {I18n('add books')}
+    </a>
+    <button
+      class="action tiny-button light-blue"
+      on:click={() => showShelfCreator = true}
+    >
+      {@html icon('plus')}{I18n('create shelf')}
+    </button>
+  {:else}
+    {#if distanceFromMainUser}
+      <button
+        class="showUserOnMap tiny-button light-blue"
+        on:click={showUserOnMap}
+      >
+        {@html icon('map-marker')}
+        <span class="label">
+          {i18n('km_away_from_you', { distance: distanceFromMainUser })}
+        </span>
+      </button>
+    {/if}
+    {#await waitingForUpdate}
+      <Spinner />
+    {:then}
+      {#if relationState === 'friends'}
+        <button
+          on:click={unfriend}
+          class="unfriend action tiny-button"
+          title={I18n('unfriend')}
+        >
+          {@html icon('minus')}<span class="label"> {I18n('unfriend')}</span>
+        </button>
+      {:else if relationState === 'userRequested'}
+        <button
+          on:click={cancelFriendRequest}
+          class="cancel action tiny-button"
+          title={I18n('cancel friend request')}
+        >
+          {@html icon('ban')}<span class="label"> {I18n('cancel request')}</span>
+        </button>
+      {:else if relationState === 'otherRequested'}
+        <button
+          on:click={acceptFriendRequest}
+          class="accept action tiny-button light-blue"
+          title={I18n('accept friend request')}
+        >
+          {@html icon('check')}<span class="label"> {I18n('confirm')}</span>
+        </button>
+        <button
+          on:click={discardFriendRequest}
+          class="discard action tiny-button"
+          title={I18n('discard friend request')}
+        >
+          {@html icon('minus')}<span class="label"> {I18n('decline')}</span>
+        </button>
+      {:else}
+        <button
+          on:click={sendFriendRequest}
+          class="request action tiny-button light-blue"
+          title={I18n('send friend request')}
+        >
+          {@html icon('plus')}<span class="label"> {I18n('add friend')}</span>
+        </button>
+      {/if}
+    {/await}
+  {/if}
+</div>
+
+{#if showShelfCreator}
+  <Modal
+    on:closeModal={() => showShelfCreator = false}
+  >
+    <ShelfEditor
+      inGlobalModal={false}
+      on:shelfEditorDone={() => showShelfCreator = false}
+    />
+  </Modal>
+{/if}
+
+<style lang="scss">
+  @import '#general/scss/utils';
+  .profile-buttons{
+    margin: 0.5em;
```

Comment:
there is an annoying case: the 600px one, which excludes both large and small screen `@media screen` below.
A way to solve this could be to 
```suggestion
    margin: 0.5em;
    margin-left: auto;
```
then to add `margin-left: 0.5em;` to small screen `.profile-buttons`
and to remove the `.profile-buttons` property from large screen

Reply:
That's correct. Although most of the modified lines in 6ca6c82 are not necessary: the problem was when 2 media queries ranges exclude a screen size. Everything is fine when there is only one media query range.
I would be for defining these `$below-...-screen` variables, but they would only be used for the specific cases of having 2 media queries in the same file.
The DX advantage of only using `$below-...` when necessary (and not systematically), is that when a developer scan the code and find a `$below-...` occurrence, it denotes a specific screen width use case, that one may be careful about updating.

Reply:
Since we disagree on the generalization of this kind or rule and to not block the PR, i propose to add `margin-left: auto;` as suggest above and move 6ca6c82 in another PR ?

Reply:
commit moved to https://github.com/inventaire/inventaire-client/pull/394